### PR TITLE
Deprecate `Util#RFC822_DATETIME_FORMATTER`

### PR DESCRIPTION
**Pull request by basil**
Jenkins exposes a `Util#RFC822_DATETIME_FORMATTER` field to provide thread-safe date/time formatting for RSS feeds. Both the interface and implementation rely on the outdated Apache Commons Lang 2 library, since the old Java `SimpleDateFormat` class was not thread-safe. New versions of Java offer an immutable and thread-safe way to do this without third-party libraries, so generate our RSS feeds with that instead.

While I could not find any usages of this field in Jenkins plugins or in CloudBees plugins, I have chosen to deprecate it rather than remove it entirely in case there are any proprietary plugins outside CloudBees that are using it. It may be removed in the future once consumers have had sufficient time to react to the deprecation warning.

The new implementation uses `RFC_1123_DATE_TIME`, which is essentially a more strict and updated version of RFC 822 that requires 4-digit years instead of permitting 2-digit or 4-digit years. We were already outputting 4-digit years, so no change.

I used `ZoneId#systemDefault` even though various pieces of advice online suggested using GMT instead. I did this to preserve existing behavior and ensure no regressions for users who are relying on RSS feeds being in their local time zone.

### Testing done

To test this, I verified that the output of `Functions#rfc822Date` was the same when running in Pacific time and passed a `Calendar` object in both GMT and Pacific time zones. This ensures the behavior is the same as before and there should be no regressions for users who are relying on RSS feeds being in their local time zone.

I also ensured that `hudson.plugins.emailext.plugins.content.ScriptContentBuildWrapperTest`, which calls this code, still passes after this PR.

I also ran `mvn clean verify -Dtest=hudson.FunctionsTest`.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]
Thanks

**Review by NotMyFault — APPROVED** [review]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
